src/lib.rs

```diff
@@ -39,6 +52,14 @@ pub enum VhalError {
     ReceiveMessageError(std::io::Error),
     #[error("Unexpected message length, expected {0}, found {1}")]
     ReceiveMessageLengthError(usize, usize),
+    #[error("Unexpected message status: {0}")]
+    ReceiveMessageStatusError(usize),
+    #[error("Unexpected message type: {0}")]
+    ReceiveMessageTypeError(usize),
+    #[error("Unexpected message value type")]
+    ReceiveMessageValueTypeError,
+    #[error("Received message invalid value")]
+    ReceiveMessageValueError,
```

Comment:
Is it worth to also add a parameter? Value type & value accordingly?

Reply:
Ideally we would have it, yes. I have been checking again, but the problem is that, on the one hand, these errors are a bit multipurposed; and on the other hand, the value is `Option`, so it is not always possible to give the error a value, and even if it is, we need to unpack it first.

I could architecture this better, but since part of the codebase is autogenerated, it is not straight forward. I'd suggest leave it as is for the time being, and if we are annoyed debugging errors from the library, we refine later.

Reply:
OK, then can resolve this one :)